game-core/src/main/java/games/strategy/net/AbstractServerMessenger.java

```diff
@@ -378,51 +374,18 @@ public boolean isUsernameMiniBanned(final String username) {
   }
 
   @Override
-  public void notifyUsernameMiniBanningOfPlayer(final String username, final @Nullable Instant expires) {
-    synchronized (cachedListLock) {
-      if (!miniBannedUsernames.contains(username)) {
-        miniBannedUsernames.add(username);
-      }
-      if (expires != null) {
-        final Timer unbanUsernameTimer = new Timer("Username unban timer");
-        unbanUsernameTimer.schedule(
-            newUnbanTimerTask(() -> miniBannedUsernames.remove(username)),
-            millisBetweenNowAnd(expires));
-      }
-    }
-  }
-
-  private TimerTask newUnbanTimerTask(final Runnable unbanUser) {
-    return new TimerTask() {
-      @Override
-      public void run() {
-        synchronized (cachedListLock) {
-          unbanUser.run();
-        }
-      }
-    };
+  public void notifyUsernameMiniBanningOfPlayer(final String username) {
+    miniBannedUsernames.add(username);
   }
 
   @Override
   public boolean isIpMiniBanned(final String ip) {
-    synchronized (cachedListLock) {
-      return miniBannedIpAddresses.contains(ip);
-    }
+    return miniBannedIpAddresses.contains(ip);
   }
 
   @Override
-  public void notifyIpMiniBanningOfPlayer(final String ip, final @Nullable Instant expires) {
-    synchronized (cachedListLock) {
-      if (!miniBannedIpAddresses.contains(ip)) {
-        miniBannedIpAddresses.add(ip);
-      }
-      if (expires != null) {
-        final Timer unbanIpTimer = new Timer("IP unban timer");
-        unbanIpTimer.schedule(
-            newUnbanTimerTask(() -> miniBannedIpAddresses.remove(ip)),
-            millisBetweenNowAnd(expires));
-      }
-    }
+  public void notifyIpMiniBanningOfPlayer(final String ip) {
+    miniBannedIpAddresses.add(ip);
```

Comment:
Shouldn't there be a lock around this action, or doo we have a guarantee all calls are coming from the same Thread?

Reply:
It's worth considering a bit more at least. I'm curious if we can see where/why there should be locking so we don't just blindly throw more locks down.

Reply:
It looks like the external synchronization is a bit old and a thread safe collection should do the trick. At the risk of blowing up scope, I updated the collection types to be a thread-safe variant and removed the external synchronization: aee0acee0da59f2c8512e34a8a7ad5be6af9c453